code/modules/food_and_drinks/food/snacks_pie.dm

```diff
@@ -43,8 +47,19 @@
 		reagents.reaction(hit_atom, TOUCH)
 	if(isliving(hit_atom))
 		var/mob/living/L = hit_atom
-		if(stunning)
-			L.Paralyze(20) //splat!
+		if(L.client && H)
+			var/experience_given = 30
+			if(L.GetComponent(/datum/component/creamed))
+				experience_given *= 0.1
+			if(HAS_TRAIT(L.mind, TRAIT_LAW_ENFORCEMENT_METABOLISM))
+				experience_given *= 3
```

Comment:
Isn't there some way to add the experience upon the component getting added/inherited.
Should probably be over there instead of here.

Reply:
i would have to pass at least 2 arguments into the component and then have component have more checks, i think its more expensive than 1 getcomponent bool  call